### DISCUSSION TITLE
Question Regarding Google Identity Services

### DISCUSSION THREAD

**Discussion by arqcenick** [pingcap/gdocwiki#160] [Q&A]
Hi, thank you for this great tool. I have a question regarding the OAuth process which is present in the current implementation.
When I set up the project and run the server, it gave an error which directs to a google page stating:

> We are [discontinuing the Google Sign-In JavaScript Platform Library for web](https://developers.googleblog.com/2021/08/gsi-jsweb-deprecation.html). The library will be unavailable for download after the March 31, 2023 deprecation date. Instead, use the new Google Identity Services for [Web](https://developers.google.com/identity/gsi/web).
By default, newly created Client IDs are now blocked from using the older Platform Library, existing Client IDs are unaffected. New Client IDs created before July 29th, 2022 can set [`plugin_name`](https://developers.google.com/identity/sign-in/web/reference) to enable use of the Google Platform Library.

Is there an easy way to update to Google Identity Services, or do you plan to update this tool in the future?

**Comment by breezewish** [comment]
Thanks for the info! I will take a look and try to update the library soon.

**Comment by adaviding** [comment]
It looks like a fix is underway here, but the PR has just sat there for 3 weeks.  Is anyone able to give it a review?  https://github.com/pingcap/gdocwiki/pull/162

(I have not done any testing on it.)
